integrations/wpseo/wpseo.php

```diff
@@ -467,7 +473,9 @@ public function copy_post_metas( $keys, $sync ) {
 	 */
 	public function translate_post_meta( $value, $key, $lang ) {
 		if ( false !== strpos( $key, '_yoast_wpseo_primary_' ) ) {
-			$value = pll_get_term( $value, $lang );
+			if ( PLL()->model->is_translated_taxonomy( $key ) ) {
+				return pll_get_term( $value, $lang );
+			}
 		}
 		return $value;
```

Comment:
`PLL()->model->is_translated_taxonomy( $key )` won't work, it's the full name of the meta with the prefix `'_yoast_wpseo_primary_'`.

```php
if ( 0 !== strpos( $key, '_yoast_wpseo_primary_' ) ) {
	return $value;
}

$taxonomy = str_replace( '_yoast_wpseo_primary_', '', $key );

if ( ! PLL()->model->is_translated_taxonomy( $taxonomy ) ) {
	return $value;
}

return pll_get_term( $value, $lang );
```